chore: perftests should use API now that `influx` cmd is gone

codyshepherd opened the pull request:
Replaces use of `influx [command]` with curls to the API, which fixes breakages to perf tests following #21797. 


<!-- Checkboxes below this note can be erased if not applicable to your Pull Request. -->

- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [x] Rebased/mergeable
- [x] Tests pass

danxmoran commented on scripts/ci/perf_test.sh:
```diff
@@ -58,18 +58,18 @@ EOF
 export TEST_ORG=example_org
 export TEST_TOKEN=token
-ssh ubuntu@$ec2_ip "influx setup -c default -b benchmark_db -n default -o $TEST_ORG -p thisisnotused -r 0 -t $TEST_TOKEN -u ubuntu --skip-verify --force"
+ssh ubuntu@$ec2_ip "curl -s -H \"Content-Type: application/json\" -XPOST -d '{\"username\": \"default\", \"password\": \"thisisnotused\", \"retentionPeriodSeconds\": 0, \"retentionPeriodHours\": 0, \"org\": \"$TEST_ORG\", \"bucket\": \"unused_bucket\", \"token\": \"$TEST_TOKEN\"}' http://localhost:8086/setup -w %{http_code}"
```
You can leave out `retentionPeriodHours` here if you want (it's deprecated / replaced by `retentionPeriodSeconds`)

danxmoran replied:
Should we check that this returns a successful code?

codyshepherd replied:
With `set -e` we "get that for free" in that the ssh wrapper will return nonzero if its command does. Which is how I found out about this breakage to begin with.

codyshepherd replied:
Yes, I'll fix.

codyshepherd replied:
Hmm, but even the following returns a 200?

`ssh -i /home/ubuntu/.ssh/key.pem ubuntu@<public IP> 'curl -o /dev/null -XPOST -H '\''application/json'\'' http://localhost:8086/nonsense -d '\''{"junk": "nothing"}'\'' -w %{http_code}'`

danxmoran replied:
That's an unfortunate side-effect of the embedded UI, I think. Other than a few one-off things like `/ready`, any route that doesn't start with `/api/v2` is assumed to be a UI route. Since the back-end doesn't know any details about the UI it just returns the base HTML page w/ a 200 response, assuming the embedded React router in the front-end will handle parsing the route.

codyshepherd replied:
Ah, so we may actually get a 50x from `/ready` even though other endpoints are set to just return 200?

danxmoran replied:
`/ready` is currently hard-coded to return a 200 from the backend, but it could hypothetically return a 50x if we ever update the impl. The call to the setup API here could also definitely return an error code.

I just noticed, the HTTP path here needs to be `/api/v2/setup` for the request to reach the back-end

codyshepherd replied:
Ok, changed

codyshepherd replied:
The [API docs state that /ready could return a non-2xx error](https://docs.influxdata.com/influxdb/v2.0/api/#operation/GetReady). It sounds like that wouldn't come from the endpoint itself but from the server in general if something were really borked

danxmoran commented on scripts/ci/perf_test.sh:
```diff
@@ -58,18 +58,22 @@ EOF
 export TEST_ORG=example_org
 export TEST_TOKEN=token
-ssh ubuntu@$ec2_ip "influx setup -c default -b benchmark_db -n default -o $TEST_ORG -p thisisnotused -r 0 -t $TEST_TOKEN -u ubuntu --skip-verify --force"
+result=$(ssh ubuntu@$ec2_ip "curl -s -H \"Content-Type: application/json\" -XPOST -d '{\"username\": \"default\", \"password\": \"thisisnotused\", \"retentionPeriodSeconds\": 0, \"retentionPeriodHours\": 0, \"org\": \"$TEST_ORG\", \"bucket\": \"unused_bucket\", \"token\": \"$TEST_TOKEN\"}' http://localhost:8086/setup -w %{http_code}")
```
Making sure it doesn't get lost from the other thread:
```suggestion
result=$(ssh ubuntu@$ec2_ip "curl -s -H \"Content-Type: application/json\" -XPOST -d '{\"username\": \"default\", \"password\": \"thisisnotused\", \"retentionPeriodSeconds\": 0, \"retentionPeriodHours\": 0, \"org\": \"$TEST_ORG\", \"bucket\": \"unused_bucket\", \"token\": \"$TEST_TOKEN\"}' http://localhost:8086/api/v2/setup -w %{http_code}")
```

codyshepherd commented:
Ok, removed the retention hours arg, and added the `api/v2/` portion of the endpoint, though when I ran this curl command manually against an influxdb2 installation, it seemed to work fine without specifying that part. Maybe I missed something.
